sonaranalyzer-dotnet/rspec/cs/S1479_c#.html

```diff
@@ -1,5 +1,55 @@
-<p>When <code>switch</code> statements have large sets of <code>case</code> clauses, it is usually an attempt to map two sets of data. A real map
-structure would be more readable and maintainable, and should be used instead.</p>
+<p>When <code>switch</code> statements have large sets of case clauses, it is usually an attempt to map two sets of data. A <code>Dictionary</code>
```

Comment:
FYI @pavel-mikula-sonarsource  @christophe-zurn-sonarsource  there's an ongoing discussion around this RSPEC on the community forum https://community.sonarsource.com/t/c-switch-too-many-case-clauses-better-suggested-fix/16993/2

From what I can tell (also see Loic's last comment/whisper), there isn't a clear agreement on what it should become.

However, I believe this form is better than before. Before it was mentioning a map (Java terms), now it's mentioning the C#-specific `Dictionary` and adding a use-case with it.

Reply:
As long as the RSPEC matches the current rule implementation, it should be fine for _this_ release.

Reply:
I agree

Reply:
Indeed, new description is more detailed and adds code example. Even though the discussion around it is still ongoing, there is no issue in updating current description, since this is already the behaviour of the rule.